src/Resources/js/modules/participation-update-handler.ts

```diff
@@ -19,192 +22,418 @@ export interface AcceptOfferData {
+        });
+        this.toggleTooltip($checkbox);
+
+        let $dishContainer = $checkbox.closest('.meal-row');
+        if (1 > $dishContainer.length) {
+            return;
+        }
 
-    public static toggle($checkbox: JQuery, data: ToggleData) {
-        // change
-        ParticipationUpdateHandler.changeCheckboxState($checkbox);
-        const nextAction = ('deleted' === data.actionText) ? ParticipationAction.JOIN : ParticipationAction.DELETE;
-        ParticipationUpdateHandler.changeCheckboxAttributes($checkbox, nextAction, data.url);
-        ParticipationUpdateHandler.changeParticipationCounter($checkbox, ParticipationState.DEFAULT, data.participantsCount);
+        if (CombinedMealService.isCombinedDish($dishContainer)) {
+            CombinedMealService.updateDishes($checkbox, data.participantID, data.bookedDishSlugs);
+        }
+    }
+
+    public static updateParticipation(data: ParticipationUpdateData[]) {
+        for (const [mealId, update] of Object.entries(data)) {
+            let $checkbox = $(`div[data-id=${mealId}] input[type=checkbox]`);
+            if (1 > $checkbox.length) {
+                console.error(`checkbox not found, mealId: ${mealId}`);
+                continue;
+            }
 
-        // update
-        ParticipationUpdateHandler.updateCheckboxEnabled($checkbox);
-        ParticipationUpdateHandler.updateCheckBoxWrapper($checkbox);
-        CombinedMealService.updateDish($checkbox, data.participantID, data.bookedDishSlugs);
+            const state = ParticipationUpdateHandler.getNextState(
+                $checkbox,
+                $checkbox.prop('checked'),
+                update.available,
+                update.locked
+            );
+            const [action, url] = ParticipationUpdateHandler.stateToAction($checkbox, state);
+            ParticipationUpdateHandler.updateStatus($checkbox, {
+                state: state,
+                count: update.count,
+                action: action,
+                actionURL: url
+            });
 
-        let $slotBox = $checkbox.closest('.meal').find('.slot-selector');
-        $slotBox.addClass('tmp-disabled').prop('disabled', true);
-        $slotBox.parent().children('.loader').css('visibility', 'visible');
+            let $mealContainer = $checkbox.closest('.meal-row');
+            if (1 > $mealContainer.length) {
+                continue;
+            }
+
+            if (update.availableWith === undefined) {
+                $mealContainer.attr('data-available-dishes', '');
+            } else {
+                $mealContainer.attr('data-available-dishes', update.availableWith.join(','))
+            }
+        }
     }
 
-    public static acceptOffer($checkbox: JQuery, data: AcceptOfferData): void {
-        this.changeToSwapState($checkbox, data.url, data.participantsCount);
+    public static changeToAssignedSlot($checkbox: JQuery, slot: string){
+        let $assignedSlot = $checkbox.closest('.meal').find('[value='+ slot +']');
```

Comment:
would write it without extra variable

$checkbox.closest('.meal')
  .find('[value='+ slot +']')
  .prop('selected', 'selected');